crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -796,6 +808,70 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
         }
     }
 
+    /// Given a candidate package and version, return its dependencies.
+    #[instrument(skip_all, fields(%package, %version))]
+    fn get_dependencies_forking(
+        &self,
+        package: &PubGrubPackage,
+        version: &Version,
+        priorities: &mut PubGrubPriorities,
+        request_sink: &Sender<Request>,
+    ) -> Result<Vec<Dependencies>, ResolveError> {
+        type Dep = (PubGrubPackage, Range<Version>);
+
+        let result = self.get_dependencies(package, version, priorities, request_sink);
+        if self.markers.is_some() {
+            return result.map(|deps| vec![deps]);
+        }
+        let deps: Vec<Dep> = match result? {
+            Dependencies::Available(deps) => deps,
+            Dependencies::Unavailable(err) => return Ok(vec![Dependencies::Unavailable(err)]),
+        };
+
+        // FIXME: This assumes that the version ranges are non-overlapping.
+        // When we do this for real, I believe it is true that we do NOT want
+        // to fork when there are overlapping version ranges.
```

Comment:
Can you include an example here?

Reply:
Separately, what _do_ we want to do? Take the intersection? What if we can't solve the intersection...? Is that possible?

Reply:
I was thinking that if we took the intersection and couldn't solve it, then the resolution overall would fail.

But now that I think about this more, I'm no longer convinced we should do this. And that checking that the markers are disjoint should be enough. I think what I had thought previously was that if the versions overlapped, then it would be possible to emit a lock file whose graph traversal would result in selecting two different versions of the same package. But I don't think that's true _given_ that we require an empty intersection for the corresponding marker expressions (which I have not added yet).

I'll remove this comment. Thanks for flagging it.